src/async_fuse/memfs/cache/memory_cache.rs

```diff
@@ -3,26 +3,139 @@
 use async_trait::async_trait;
 use clippy_utilities::OverflowArithmetic;
 use lockfree_cuckoohash::{pin, LockFreeCuckooHash as HashMap};
-use std::collections::HashMap as StdHashMap;
+use std::collections::{BTreeSet, HashMap as StdHashMap};
+use std::num::NonZeroUsize;
 use std::sync::Arc;
-use tokio::sync::RwLock;
+use std::time::Duration;
+use tokio::sync::{mpsc, oneshot, RwLock};
+use tracing::warn;
 
 use super::policy::EvictPolicy;
-use super::{Block, BlockCoordinate, BlockId, Storage};
+use super::{write_back_task, Block, BlockCoordinate, BlockId, Command, SoftLimit, Storage};
 use crate::async_fuse::fuse::protocol::INum;
 
-/// The file-level cache.
-type FileCache = Arc<RwLock<StdHashMap<BlockId, Block>>>;
-
 /// Merge the content from `src` to `dst`. This will set `dst` to be dirty.
 fn merge_two_blocks(src: &Block, dst: &mut Block) {
-    dst.set_dirty();
+    dst.set_dirty(true);
     dst.update(src);
 }
 
+/// A builder to configure and build a `MemoryCache`.
+pub struct MemoryCacheBuilder<P, S> {
```

Comment:
This should be moved to another file, as each file should maintain its own minimal responsibility.

Reply:
`MemoryCacheBuilder` is highly related with `MemoryCache`, so I insist on leaving it in the same file.

Reply:
Then, let's put them in a subfolder named `memory_storage` or something, with files like `mod.rs`, `memory_cache.rs`, `build.rs`, and `test.rs`. This file is already 1000 lines long.

Reply:
Tests for other modules might also need to be split. For example, `storage_manager` has 700 lines of tests and 300 lines of source code.

Reply:
> Then, let's put them in a subfolder named `memory_storage` or something, with files like `mod.rs`, `memory_cache.rs`, `build.rs`, and `test.rs`. This file is already 1000 lines long.

This is a good idea， I will handle it later. However, I think it's not good to handle `storage_manager` related issues in this PR.